clang/lib/Sema/SemaConcept.cpp

```diff
@@ -1519,7 +1519,8 @@ bool Sema::IsAtLeastAsConstrained(NamedDecl *D1,
     auto IsExpectedEntity = [](const FunctionDecl *FD) {
       FunctionDecl::TemplatedKind Kind = FD->getTemplatedKind();
       return Kind == FunctionDecl::TK_NonTemplate ||
-             Kind == FunctionDecl::TK_FunctionTemplate;
+             Kind == FunctionDecl::TK_FunctionTemplate ||
```

Comment:
This does not seem correct as it would make the assert useless, right?

I think the cases that should be valid are TK_NonTemplate, TK_MemberSpecialization, TK_FunctionTemplateSpecialization. (ie the non-dependent cases)

Reply:
Seems right - Done; tests still seem to pass.